radixdlt/babylon-wallet-android
# [ABW-2429] improve mnemonic repository error handling

## Description
This PR is supposed to improve app behavior when installed in samsung secure folder, however it does not entirely fix issues users might have using app in samsung secure folder. It should prevent app from crashing, but not necesarilly make it usable since we still didn't get to the bottom of why app does not work in secure folder. Changes:
- don't assume mnemonic is not null in flows where we need it. If mnemonic exist, but we can't read it because of encryption issues (like in secure folder), throw error message.
- removed mandatory main babylon factor source recovery. I've added NoMnemonic exception being thrown in account/persona creation and in the account recovery scan, so we no longer need to force user to restore mnemonic at app startup, he can do it at later time.
- detect error when creating first account, which can be linked to samsung secure folder, but didn't yet add any specific message for it - right now it is just a toast. 
Weird findings about secure folder:
- when app is in secure folder, it looks like biometrics authentication with FINGERPRINT simple does not grant us access to generated encryption key that is stored in keystore. When we generate key, we set authentication type to be `KeyProperties.AUTH_BIOMETRIC_STRONG or KeyProperties.AUTH_DEVICE_CREDENTIAL`. It looks like in this scenario fingerprint is not considered by a device BIOMETRIC_STRONG class when app is run in secure folder. I was reading method by method documentation of all the calls we use to setup key generation, but was not able to pin down why this happens.

## How to test

1. Run app in samsung secure folder and try to authenticate with biometrics, and later with PIN and biometrics afterwards
2. Run app on any device, change security settings and observe that you are no longer prompted to recover BDFS. Then try to do account recovery/sign tx/create account/create persona and verify that app does not crash in any of those cases, and throws error instead.
